src/java.base/share/classes/jdk/internal/misc/InnocuousThread.java

```diff
@@ -124,26 +124,38 @@ public Thread run() {
      */
     public static Thread newSystemThread(String name, Runnable target,
                                          long stackSize, int priority) {
+        return newSystemThread(name, target, stackSize, priority, false);
+    }
+
+    /**
+     * Returns a new InnocuousThread with null context class loader.
+     * Thread priority is set to the given priority.
+     * If {@code daemon} is {@code true}, the thread is set as a daemon thread.
+     */
+    public static Thread newSystemThread(String name, Runnable target,
+                                         long stackSize, int priority,
+                                         boolean daemon) {
```

Comment:
Thanks for adding this overload, I think that it will be useful for the future too.   ( it never seems to matter how many variants of this factories we have, we still need one more :-) )

Reply:
I would prefer to to avoid creating new factories when the desired function can be done on the resulting thread.
Such as `setDaemon()` and `setName()`, etc.
It does avoid the doPriv in this case, but is not necessary and when the security manager goes away, will leave around clutter (duplicated) functionality.

Reply:
Looking beyond this specific change, there is a lot of potential use for this new factory elsewhere in the code. It also avoids similar bugs from possibly reoccurring (by having the setDaemon call inside the factory).

Reply:
In the interest of making progress, let’s revert the change to the factory. This can be done separately, if at all.